### ISSUE TITLE
Rule suggestion: no negated condition in if-else

### ISSUE THREAD

**Issue by adamsol** [astral-sh/ruff#24249]
### Summary

See [this ESLint rule](https://eslint.org/docs/latest/rules/no-negated-condition) for a detailed description.

There's already something similar in Ruff: https://docs.astral.sh/ruff/rules/if-expr-with-twisted-arms/, but much more narrow in scope - works only for `b if not a else a`, not for `b if not a else c` and not for `if` statements.

**Comment by ntBre** [comment]
Thanks for the suggestion! I was surprised I couldn't find a previous request for this. I think the analogous clippy rule is [`if_not_else`](https://rust-lang.github.io/rust-clippy/master/?groups=cargo%2Ccomplexity%2Ccorrectness%2Cnursery%2Cpedantic%2Cperf%2Crestriction%2Cstyle%2Csuspicious%2Cdeprecated&search=negat#if_not_else).

It would flag code like this:

```py
def foo(condition):
    if not condition:
        return 1
    else:
        return 2
```

and suggest:

```py
def foo(condition):
    if condition:
        return 2
    else:
        return 1
```

if I understand correctly.

We actually disable the clippy version in our own config, but that's not necessarily a reason to avoid having the rule for those who do like it 😆 

https://github.com/astral-sh/ruff/blob/29bf84e064567bde67320efd164a02b7322cc664/Cargo.toml#L272-L279

**Comment by adamsol** [comment]
Actually, this was kinda already requested as [WPS504](https://wemake-python-styleguide.readthedocs.io/en/latest/pages/usage/violations/refactoring.html#wemake_python_styleguide.violations.refactoring.NegatedConditionsViolation) in #3845. Not the easiest to find, but feel free to close this issue if it's considered a duplicate.

**Comment by gamer191** [comment]
> ```py
> def foo(condition):
>     if not condition:
>         return 1
>     else:
>         return 2
> ```

This is an interesting example, because [RET505](https://docs.astral.sh/ruff/rules/superfluous-else-return/) would change it to

> ```py
> def foo(condition):
>     if not condition:
>         return 1
>     return 2
> ```

And then the proposed rule presumably wouldn't detect that it can be optimised to:

> ```py
> def foo(condition):
>     if condition:
>         return 2
>     return 1
> ```

**Comment by VascoSch92** [comment]
I think this rule is somewhat opinionated. It's not always clear what the best way to write an if-else block is. 

Sometimes a negative condition reads more idiomatically than a positive one. It's really a matter of taste.

That said, I'm not saying the rule isn't useful.

Just my 2 cents.

**Comment by adamsol** [comment]
For me, it's the double negation that makes a condition harder to grasp. I've also seen arguments that the first branch should represent the "more frequent" case, but that seems to be more subjective, since it's not always clear which case will be more frequent or idiomatic. This rule simply removes the dilemma and keeps all conditions more consistent.

Here's some blog post and additional comments about this: https://dev.to/somedood/please-dont-write-confusing-conditionals-2n32

Edit: [ast-grep](https://ast-grep.github.io/) rule for this feature, in case anyone is interested:
```
language: python
message: negated condition in if-else
rule:
  any:
    - pattern: $A if $COND else $B
    - pattern: |
        if $COND:
            $A
        else:
            $B
constraints:
  COND:
    any:
      - pattern: not $X
      - pattern: $X != $Y
      - pattern: $X is not $Y
      - pattern: $X not in $Y
```
